src_c/_sdl2/controller.pyx

```diff
@@ -0,0 +1,129 @@
+from . import error
+
+cdef extern from "../pygame.h" nogil:
+    int pgJoystick_Check(object joy)
+    object pgJoystick_New(int);
+    void import_pygame_joystick()
+	
+import_pygame_joystick()
+
+cdef extern from "SDL.h" nogil:
+    void SDL_free(void *mem)
+
+def set_eventstate(state):
+    return SDL_GameControllerEventState(state)
+
+def get_count():
+    """ Returns the number of attached joysticks.
+    """
+    num = SDL_NumJoysticks()
+    if num < 0:
+        raise error()
+    return num
+
+def update():
+    """ Will automatically called by the event loop,
+        not necessary to call this function.
+    """
+    SDL_GameControllerUpdate()
+
+def is_controller(index):
+    """ Check if the given joystick is supported by the game controller interface.
+
+    :param int index: Index of the joystick.
+
+    :return: 1 if supported, 0 if unsupported or invalid index.
+    """
+    return SDL_IsGameController(index)
+
+def name_forindex(index):
+    """ Returns the name of controller,
+        or NULL if there's no name or the index is invalid.
+    """
+    return SDL_GameControllerNameForIndex(index)
+
+cdef class Controller:
```

Comment:
I think you forgot to close it in a __dealloc__ method.

Reply:
For some reason,

```
def __dealloc__(self):
        if self._controller:
            SDL_GameControllerClose(self._controller)
```

caused weird behavior on exit(no error in Python Shell, but looks like some error occurred running cython code.

but closing controllers anytime before exit runs fine.